addons/stock_account/models/product.py

```diff
@@ -113,7 +142,29 @@ class ProductProduct(models.Model):
     def write(self, vals):
         if 'standard_price' in vals and not self.env.context.get('disable_auto_svl'):
             self.filtered(lambda p: p.cost_method != 'fifo')._change_standard_price(vals['standard_price'])
-        return super(ProductProduct, self).write(vals)
+        return super().write(vals)
+
+    @api.onchange('standard_price')
+    def _onchange_standard_price(self):
+        if self.lot_valuated:
+            return {
+                'warning': {
+                    'title': _("Warning"),
+                    'message': _("This product is valuated by lot/serial number. Changing the cost \
+will update the cost of every lot/serial number in stock."),
+                }
+            }
+
+    @api.onchange('lot_valuated')
+    def _onchange_lot_valuated(self):
```

Comment:
Strange to have it in product.product but not in template while the field is in template